Model/Api/CreateOrder.php

```diff
@@ -558,30 +558,32 @@ public function hasItemErrors($quoteItem)
      * @param array     $transactionItems
      * @throws BoltException
      */
-    public function validateItemPrice($itemSku, $itemPrice, $transactionItems)
+    public function validateItemPrice($itemSku, $itemPrice, &$transactionItems)
     {
-        foreach ($transactionItems as $transactionItem) {
+        foreach ($transactionItems as $index => $transactionItem) {
             $transactionItemSku = $this->getSkuFromTransaction($transactionItem);
             $transactionUnitPrice = $this->getUnitPriceFromTransaction($transactionItem);
 
-            if ($transactionItemSku === $itemSku
-                && abs($itemPrice - $transactionUnitPrice) > OrderHelper::MISMATCH_TOLERANCE
+            if ($transactionItemSku === $itemSku &&
+                abs($itemPrice - $transactionUnitPrice) <= OrderHelper::MISMATCH_TOLERANCE
             ) {
-                $this->bugsnag->registerCallback(function ($report) use ($itemPrice, $transactionUnitPrice) {
-                    $report->setMetaData([
-                        'Pre Auth' => [
-                            'item.price' => $itemPrice,
-                            'transaction.unit_price' => $transactionUnitPrice,
-                        ]
-                    ]);
-                });
-                throw new BoltException(
-                    __('Price does not match. Item sku: ' . $itemSku),
-                    null,
-                    self::E_BOLT_ITEM_PRICE_HAS_BEEN_UPDATED
-                );
+                unset ($transactionItems[$index]);
```

Comment:
Why we should unset some item?

Reply:
@YevhenShBolt if we don't we could have false positive results. Here we match quote items to bolt transaction items. Theoretically two or more quote items with the same SKU and price (and some different options) could be matched to a single one item within the bolt transaction in this loop. We could solve this differently, for example add another loop and match them in the opposite direction, but this way it is more optimal and simpler.